pkg/kubestatemetrics/builder/builder.go

```diff
@@ -228,7 +228,24 @@ func (b *Builder) startReflector(
 	expectedType interface{},
 	store cache.Store,
 	listWatcher cache.ListerWatcher,
+	useApiServerCache bool,
 ) {
+	if useApiServerCache {
+		listWatcher = &cacheEnabledListerWatcher{ListerWatcher: listWatcher}
+	}
 	reflector := cache.NewReflector(listWatcher, expectedType, store, b.resync*time.Second)
 	go reflector.Run(b.ctx.Done())
 }
+
+type cacheEnabledListerWatcher struct {
+	cache.ListerWatcher
+}
+
+// List uses a resource version of "0" to list all resources.
+// The implementation is required to avoid a quorum from ETCD.
+// https://github.com/kubernetes/kubernetes/blob/dae37c21645c0fc04bbb0202ad1eba6ba4438d23/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L526-L527
+// Refer to https://github.com/kubernetes/kube-state-metrics/blob/7995d5fd23bcff7ae24ab6849f7c393d262fb025/pkg/watch/watch.go#L77
+func (c *cacheEnabledListerWatcher) List(options v1.ListOptions) (runtime.Object, error) {
+	options.ResourceVersion = "0"
```

Comment:
I used a debuglog to make sure it's set: `2024-06-05 13:25:33 UTC | CLUSTER | INFO | (pkg/kubestatemetrics/builder/builder.go:249 in List) | debug rv set`